lib/datadog/core/event.rb

```diff
@@ -0,0 +1,99 @@
+# typed: false
+require 'datadog/core/logger'
+
+module Datadog
+  module Core
+    # Event behavior and DSL
+    module Events
```

Comment:
Hmmm, does this make sense to have in the core? Looking at the classes that we needed to change to move this, all of them seem to be in the tracer.

Have you considered moving this inside the tracing namespace? We could later move it to the core when we needed it, but until then we would avoid having functionality that is not being shared yet there.

I suggest this because I don't think the core should become an "utils" package: a place where we accumulate things over time that could be shared. For instance, the profiler also has its own `Datadog::Profiling::Event` class and `Datadog::Profiling::Events` namespace that are completely unrelated to this one.

Reply:
> I suggest this because I don't think the core should become an "utils" package: a place where we accumulate things over time that could be shared.

I think this is a fair question about direction or general rule of thumb. I also don't want to clutter a new package with old stuff it doesn't need. I could be convinced this is the way to go. That said, if we go this way, would the following be the rules we'd adopt?

> 1. If a component is referenced by components belonging to only one product, it belongs in that product.
> 2. If a component is referenced by components belonging to two or more different products, it belongs in `Core`.